node/src/components/consensus/highway_core/active_validator/round_success_meter.rs

```diff
@@ -0,0 +1,100 @@
+use std::collections::VecDeque;
+
+use crate::types::Timestamp;
+
+const NUM_ROUNDS_TO_CONSIDER: usize = 40;
+const MAX_FAILED_ROUNDS: usize = 10;
+const ACCELERATION_PARAMETER: usize = 1000;
```

Comment:
This won't work with our short eras.
We need to make sure that at least once per era, we have a chance to accelerate.
On the other hand we don't want to always accelerate once per era, and always go way too fast; so I think we shouldn't accelerate _unconditionally_ anymore. Maybe only if there are fewer than, say, 5 failed rounds?

Another option would be to somehow carry over the success meter from one era to the next… :grimacing:

Reply:
Hmm... Decelerating won't work either with eras that have 10 blocks each (in tests). I think carrying over the result would be the best way, but this would also require us to change the way round exponent is initialized in each era, so that it won't change between eras in most cases (so it couldn't be initialized to the minimum value every time).

Reply:
Per conversations on Slack, I went with conditional acceleration (conditional on there being few enough failures among last x rounds), which should work slightly better, but additional work is still needed and ticketed on JIRA.